Assignment/Assignment/SampleData.cs

```diff
@@ -1,30 +1,105 @@
 ﻿using System;
 using System.Collections.Generic;
+using System.Linq;
+using System.IO;
+using System.Diagnostics.CodeAnalysis;
 
 namespace Assignment
 {
     public class SampleData : ISampleData
     {
         // 1.
-        public IEnumerable<string> CsvRows => throw new NotImplementedException();
+        public IEnumerable<string> CsvRows // = the current return ? 2/17 lecture
+        {
+            get
+            {
+                return File.ReadAllLines(FilePath).Skip(1).ToList();//System.Reflection.Assembly.GetExecutingAssembly().Location; current will work but it relies on unsafe assumptions
+            }
+        }
+
+        private string FilePath;
+
+        public SampleData()
+        {
+            FilePath = "People.csv";
+        }
+
+        public SampleData([DisallowNull] string filePath)
+        {
+            FilePath = filePath;
+        }
 
         // 2.
-        public IEnumerable<string> GetUniqueSortedListOfStatesGivenCsvRows() 
-            => throw new NotImplementedException();
+        public IEnumerable<string> GetUniqueSortedListOfStatesGivenCsvRows()
+        {
+
+            // Define Query
+            IEnumerable<string> stateQuery = (
+                from line in CsvRows
+                let columnSeparatedLines = line.Split(',')
+                orderby columnSeparatedLines[6]
+                select columnSeparatedLines[6]
+                ).Distinct().ToList();

```

Comment:
I would suggest sticking with the LINQ method syntax, such as below.
```suggestion
            return CsvRows
                .Select(row => row.Split(',')[6])
                .OrderBy(state => state)
                .Distinct();
```